TLM/TLM/UI/SubTools/VehicleRestrictionsTool.cs

```diff
@@ -120,15 +139,57 @@ public VehicleRestrictionsTool(TrafficManagerTool mainTool)
             // ShowSigns(false);
         }
 
+        private static NetLane[] laneBuffer => NetManager.instance.m_lanes.m_buffer;
+
+        /// <summary>
+        /// highlights the given lane according.
+        /// the highlight is emboldened if mouse click is pressed.
+        /// </summary>
+        private void RenderLaneOverlay(RenderManager.CameraInfo cameraInfo, uint laneId) {
+            var marker = new SegmentLaneMarker(laneBuffer[laneId].m_bezier);
+            bool pressed = Input.GetMouseButton(0);
+            Color color = HighlightColor;
+            color = pressed ? Color.magenta : color;
+            marker.RenderOverlay(cameraInfo, color, pressed);
+        }
+
+        /// <summary>
+        /// highlitghts all the lanes with the same sorted index as the current lane.
+        /// </summary>
+        private void RenderRoadLane(RenderManager.CameraInfo cameraInfo) {
+            SegmentLaneTraverser.Traverse(
+                renderData_.segmentId,
```

Comment:
@kvakvs Look where renderData_ is assigned. It is necessary.
What do you exactly want @kvakvs ? 
Do you want me to avoid using `renderData_` as a field altogether (which is not possbile)?
Do you want `public override void RenderOverlay(RenderManager.CameraInfo cameraInfo)` to read `renderData_.segmentId` and pass it as an argument to `RenderRoadLane` ?

Reply:
If that renderdata lives only for 1 frame, I want (if possible and doesn't break many things) to have that renderdata passed into all render funs and then get destroyed.
If it lives for more than 1 frame, disregard my comment, its all good.

Reply:
I don't understand the threading situation in CS but it is assigned by `OnToolGUI()` and then used by `RenderOverlay()` so I guess its all good.